render.go

```diff
@@ -28,20 +37,61 @@ func Render(tmpl string, vars map[string]string) (string, error) {
 	t.Option("missingkey=error")
 	var b bytes.Buffer
 	if err := t.Execute(&b, vars); err != nil {
-		return b.String(), err
+		return b.String(), secretUsed, err
 	}
 	// need to replace blank lines because of bad template formating
-	return strings.Replace(b.String(), "\n\n", "\n", -1), nil
+	return strings.Replace(b.String(), "\n\n", "\n", -1), secretUsed, nil
+}
+
+// secret generate a secret
+func secret(stringType string, length int) string {
```

Comment:
Would be good to extend the test cases for this a little more, validating the generation of secrets, length & type based on definition in template etc.